src/main/java/com/zakem/RecipeCatalog/controller/IngredientController.java

```diff
@@ -0,0 +1,41 @@
+package com.zakem.RecipeCatalog.controller;
+
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.zakem.RecipeCatalog.object.Ingredient;
+import com.zakem.RecipeCatalog.repository.IngredientRepository;
+
+@RestController
+@RequestMapping("/ingredient")
+public class IngredientController {
+	
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+	
+	IngredientRepository ingredientRepository;
+	
+	public IngredientController(IngredientRepository ingredientRepository) {
+		this.ingredientRepository = ingredientRepository;
+	}
+	
+	@PostMapping("/add")
+	public ResponseEntity<?> addIngredient(@RequestBody Ingredient ingredient){
+		Optional<Ingredient> ingredientOptional = this.ingredientRepository.findByName(ingredient.getName());
+		if (ingredientOptional.isPresent()) {
+			logger.info("Adding ingredient {} failed because already exists", ingredient.getName());
```

Comment:
Right now there's no security on your endpoint. But once you add that you will want to log _who_ made the bad request.